centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -3112,3 +3113,276 @@ function sanitizeFormHostParameters(array $ret): array
     }
     return $bindParams;
 }
+
+/**
+ * Create a new host from formData.
+ *
+ * @param array<mixed> $ret
+ *
+ * @return int|null
+ */
+function insertHostInAPI(array $ret = []): int|null
+{
+    global $centreon, $form, $isCloudPlatform, $basePath;
+
+    /** @var array<string,int|string|null> $formData */
+    $formData = !count($ret) ? $form->getSubmitValues() : $ret;
+
+    try {
+        $hostId = insertHostByApi($formData, $isCloudPlatform, $basePath);
+
+        if ((int) $formData['host_register'] === 0) {
+            updateHostTemplateService($hostId);
+        }
+
+        if (! $isCloudPlatform) {
+            if ((int) $formData['host_register'] !== 0) {
+                updateHostHostParent($hostId, $formData);
+                updateHostHostChild($hostId);
+            }
+            updateHostContactGroup($hostId, $formData);
+            updateHostContact($hostId, $formData);
+        }
+
+        if (
+            ! empty($formData['dupSvTplAssoc']['dupSvTplAssoc'])
+            || $isCloudPlatform === true
+        ) {
+            createHostTemplateService($hostId);
+        }
+
+        // Update conf change flag for poller
+        signalConfigurationChange('host', $hostId);
+        // Update host ACLs
+        $centreon->user->access->updateACL([
+            'type' => 'HOST',
+            'id' => $hostId,
+            'action' => 'ADD',
+            'access_grp_id' => (isset($formData['acl_groups']) ? $formData['acl_groups'] : null),
+        ]);
+        // Insert change logs
+        $fields = CentreonLogAction::prepareChanges($formData);
+        $centreon->CentreonLogAction->insertLog(
+            "host",
+            $hostId,
+            CentreonDB::escape($formData["host_name"]),
+            "a",
+            $fields
+        );
+
+        return ($hostId);
+    } catch (\Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . "</div>";
+
+        return (null);
+    }
+}
+
+/**
+ * Make the API request to create a new host and return the new ID.
+ *
+ * @param array $formData
+ * @param bool $isCloudPlatform
+ * @param string $basePath
+ *
+ * @throws \LogicException
+ * @throws \Exception
+ *
+ * @return int
+ */
+function insertHostByApi(array $formData, bool $isCloudPlatform, string $basePath): int
+{
+    $kernel = \App\Kernel::createForWeb();
+    /** @var Core\Infrastructure\Common\Api\Router $router */
+    $router = $kernel->getContainer()->get(Core\Infrastructure\Common\Api\Router::class)
+        ?? throw new LogicException('Router not found in container');
+    /** @var Symfony\Component\HttpClient\CurlHttpClient */
+    $client = $kernel->getContainer()->get(Symfony\Component\HttpClient\CurlHttpClient::class)
+        ?? throw new LogicException('CurlHttpClient not found in container');
```

Comment:
If you not getting the HttpClientInterface by the container, you can instanciate CurlHttpClient directly